pkg/collector/containerlogs_containerd_collector_test.go

```diff
@@ -0,0 +1,198 @@
+package collector
```

Comment:
💡 Test should reside in `tests` folder, also the whole point of unit tests is to test the "external interface" (ie public functions) to your package. That is unit tests should always be "white box" tests (see White Box Testing). That way you can refactor your code and your tests won't break.

Moving this to `tests` folder not only makes it structured as well.

Reply:
💡 **Possible solution\proposal:** 

The way these new `private scooped functs` are written seems like they are supporting `helper` to the specific package, so if we really wan to to test these private functions, we can do so by structure and scope change.

* Utils -- is a place of utility pkgs - so make that `containers-helper.go` under `utils` that way all these private functs which needs to be unit-tested will become `public` scope and as a separate unit.
* under `tests` folder there will be tests for these new `specific-container_helper_test.go` which can test these methods.
* Also it will be good idea to test `Collector` but I would recommend only write test for them if there is genuinely we can test anything, i.e. mocking something will not help if it by passes bunch of stuff.

Reply:
💡 I am fine with location et. al. but I think you just need to rationalize for these tests for Private methods, in case you absolutely fine with it, I am happy to +1 this, but why not add test for public method? Or is it something you tried already? (I do know bunch of these legacy collectors are hard to test)

Reply:
Sorry I missed this earlier - I think it is incorrect to consider these as class private methods for the purpose of whether to unit test them - they are package internal, or "non-exported" methods accessible anywhere within the "collector" package and from what I have seen it is conventional golang to unit test them like this.

Agree that we should look to add some general tests for "Collect" at the public interface level too, and agree with the caveat to be careful of mocking out the useful parts.

I had a look at adding equivalent tests for the exported (public) method instead but the amount of additional complexity or mocking required to get the same level of test coverage as we have currently put me off doing this. Again this collector will be being refactored soon to use a volume mount and potentially a different internal structure due to needing to target pod logs, so I didn't want to over commit.

Potentially the concept of a "selector" is common enough across a range of collectors that we could pull it out into being its own type, but can look at that later too.